richtextfx/src/main/java/org/fxmisc/richtext/model/SegmentOps.java

```diff
@@ -0,0 +1,42 @@
+package org.fxmisc.richtext.model;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+
+/**
+ * Defines the operations which are supported on a specific segment type.
+ *
+ * @param <SEG>
+ * @param <S>
+ */
+public interface SegmentOps<SEG, S> {
+    public int length(SEG seg);
+
+    public char charAt(SEG seg, int index);
+
+    public String getText(SEG seg);
```

Comment:
Why don't we make `getText()` and `length()` default methods that return the single `/uffc` (or whatever it is) character and a length of 1? Then, only text-related custom objects need to override that, which is expected anyway.